kratos/python_interface/__init__.py

```diff
@@ -12,6 +12,27 @@
 # import core library (Kratos.so)
 from Kratos import *
 
+def __ModuleInitDetail():
+    """
+    Configure the parallel environment.
+    This is done before initializing the kernel to ensure that MPI
+    and the parallel DataCommunicator are initialized when the Kernel is built.
+    It is defined as a function to avoid polluting the Kratos namespace with local variables.
+    """
+    if "--using-mpi" in sys.argv[1:]:
+        try:
+            import KratosMultiphysics.mpi
+        except ModuleNotFoundError:
```

Comment:
is this the same as ImportError?

Reply:
It means that it could not find the `__init__.py` of the `mpi` module (or a `mpi.py` file). `ImportError` you'd get of the file exists but then tries to import a wrong `.so`.

Reply:
thx